snuba/datasets/events_processor_base.py

```diff
@@ -221,18 +221,17 @@ def process_insert(
         stacks = exception.get("values", None) or []
         self.extract_stacktraces(processed, stacks)
 
-        if metadata is not None:
-            processed["offset"] = metadata.offset
-            processed["partition"] = metadata.partition
-            processed["message_timestamp"] = metadata.timestamp
+        processed["offset"] = metadata.offset
+        processed["partition"] = metadata.partition
+        processed["message_timestamp"] = metadata.timestamp
```

Comment:
The way these index assignments are laid out makes me wonder how much time we're spending unnecessarily resizing dictionaries. I haven't benchmarked it, but I suspect this would be more efficient with `processed.update()` or possibly using `ChainMap` (or ideally replacing these dictionaries with predefined data structures that use `__slots__`.)